### PR TITLE
New script for 389DS cn=monitor

### PR DESCRIPTION
**Pull request by sophieqc**
Hello,

This is an adaptation of the cn=monitor script for 389DS, it has been tested on FreeIPA only, but should work the same way on bare 389DS.

Thanks

### CODE REVIEW

File: check_ldap_monitor_389ds.pl [hunk 1]
```diff
@@ -0,0 +1,607 @@
+#! /usr/bin/perl -w
+
+#==========================================================================
+# Summary
+#==========================================================================
+# Check LDAP query
```
Comment (coudot) [inline check_ldap_monitor_389ds.pl]: Change the summary here to "Check 389DS monitor"

File: check_ldap_monitor_389ds.pl [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+
+#==========================================================================
+# Exit with Nagios codes
+#==========================================================================
+
+# Prepare PerfParse data
+my $perfparse = " ";
+if ($perf_data) {
+    $perfparse = "|'$type'=" . $value . ";$warning;$critical";
+}
```
Comment (coudot) [inline check_ldap_monitor_389ds.pl:572]: If $warning and $critical are not defined, you should set them to 0 to respect perfparse format. Maybe do this after the test to check their value.

**Review by coudot — CHANGES_REQUESTED** [review]

File: check_ldap_monitor_389ds.pl (hunk 1 above)
Comment (sophieqc) [reply check_ldap_monitor_389ds.pl]: Done 7521371daa8421031c85df0c6884f97303a42757

File: check_ldap_monitor_389ds.pl (hunk 2 above)
Comment (sophieqc) [reply check_ldap_monitor_389ds.pl:572]: Does ee4e30945b32ab81d1e890e46b62955c31d1fb5c looks good to you ?

Comment (sophieqc) [reply check_ldap_monitor_389ds.pl:572]: Fixed typo in df0293f58f5571e1a15b426ac3ab5e0c07276425

**Review by coudot — APPROVED** [review]

**Comment by coudot** [comment]
We will need some documentation here: https://ltb-project.org/documentation/nagios-plugins/check_ldap_monitor_389ds

**Comment by sophieqc** [comment]
I can take care of this, I'll need an account on ltb wiki or I can submit something here directly, tell me what you prefer

**Comment by coudot** [comment]
You should have received a mail with your account on the wiki

**Comment by sophieqc** [comment]
And we have documentation : https://ltb-project.org/documentation/nagios-plugins/check_ldap_monitor_389ds

**Comment by coudot** [comment]
Thanks a lot!

**Pull request merged**